troi/tools/utils.py

```diff
@@ -18,3 +21,142 @@ def create_http_session():
     http.mount("https://", adapter)
     http.mount("http://", adapter)
     return http
+
+class SoundCloudException(Exception):
+    def __init__(self, code, msg):
+        self.code = code
+        self.msg = msg
+        super().__init__(f"http error {code}: {msg}")
+
+class SoundcloudAPI:
+    def __init__(self, access_token):
+        self.access_token = access_token
+        self.headers = {
+            "Authorization": f"OAuth {self.access_token}",
+            "Content-Type": "application/json"
+        }
+        self.session = self._create_http_session()
+
+    def _create_http_session(self):
+        """ Create an HTTP session with retry strategy for handling rate limits and server errors.
+        """
+        retry_strategy = Retry(
+            total=3,
+            status_forcelist=[429, 500, 502, 503, 504], 
+            allowed_methods=["HEAD", "GET", "OPTIONS"], 
+            backoff_factor=1
+        )
+
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        http = requests.Session()
+        http.mount("https://", adapter)
+        http.mount("http://", adapter)
+
+        def raise_for_status_hook(response, *args, **kwargs):
+            try:
+                response.raise_for_status()
+            except HTTPError as http_err:
+                print(f"HTTP error occurred: {http_err}")
```

Comment:
Create a `logger = logging.getLogger(__name__)` at the top of the file and use `logger.error` here.